### ISSUE TITLE
Gridmaps do not render when using Rooms and Portals System

### ISSUE THREAD

**Issue by GDelavy** [godotengine/godot#60833]
### Godot version

3.4.2 stable official (45eaa2daf)

### System information

Windows 10, GLES 3, Nvidia Geforce 1660 Ti (30.0.14.7168)

### Issue description

- Gridmaps aren't rendered when placed inside of a Room node after calling the RoomManager's room_convert() function, because they do not inherit from CullingInstance. They remain thus invisible.

### Steps to reproduce

- Download and launch the project below.
- Launch the World scene and notice that the white surface (gridmap) doesn't render.

### Minimal reproduction project

[Gridmap Culling.zip](https://github.com/godotengine/godot/files/8643178/Gridmap.Culling.zip)

**Comment by lawnjelly** [comment]
Yes, that's a known limitation, gridmaps aren't currently supported (unless you can e.g. bake them out to regular nodes).

**Comment by GDelavy** [comment]
Thank you for the quick reply,

Is there any way to tell the Portal / Rooms system to ignore gridmaps? Even if I put the gridmap node outside of my room node, it doesn't render.

Alternatively, I've tried using the Occluder node to surround my gridmap, but the only OccluderShape available seems to be spheres. Is there a way for me to add cubes or planes or convex shapes instead?

I apologize for the many questions, but my entire project relies on gridmaps and I really don't know what I'm going to do if I don't find a workaround

**Comment by Calinou** [comment]
> Alternatively, I've tried using the Occluder node to surround my gridmap, but the only OccluderShape available seems to be spheres. Is there a way for me to add cubes or planes or convex shapes instead?

[3.5beta4](https://godotengine.org/article/dev-snapshot-godot-3-5-beta-4) has occluder polygons available, so you can use it to represent quads, boxes and the like.

**Comment by GDelavy** [comment]
Thank you so much for all the help, I really appreciate all the work you do! 

I'm going to look into implementing Lod and also experiment with 3.5beta4 to see if occluders make a significant difference. Thank you again!

**Comment by Calinou** [comment]
Reopening, as this limitation should still be documented.

**Comment by lawnjelly** [comment]
To elaborate a little - the problem stems because the gridmap is rather special - it completely overrides the scene tree, and does everything itself internally. Whereas the room conversion code is based on, and assumes, a scene tree structure. So gridmap would probably need custom conversion code written to traverse it during the room conversion, and additionally it has no machinery for setting the portal mode. There would likely be other problems as a result of it not being part of the scene tree. This is not trivial, it may represent weeks of work, and would require testing etc.

A far more practical approach would be to simply bake the gridmap into normal scene tree nodes containing meshes and multimeshes. The result could then be processed as normal during room conversion. I don't know whether this functionality exists already, or perhaps someone enthusiastic could write this. But this is not really a "portals" problem, it could probably be written by anyone who is familiar or who wants to get familiar with gridmap.
